### PR TITLE
ESDTTransferRole docs

### PR DESCRIPTION
**Pull request by dragos-rebegea**
#### Description of the pull request (what is new / what has changed)
Documentation for ESDTTransferRole

#### Did you test the changes locally ?
- [x] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [x] document new feature
- [ ] update documentation that is not relevant anymore
- [ ] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

**Review by dorin-iancu — APPROVED** [review]

File: docs/tokens/esdt-tokens.md [hunk 1]
```diff
@@ -624,6 +624,30 @@ RolesAssigningTransaction {
 
 _For more details about how arguments have to be encoded, check [here](/developers/sc-calls-format)._
 
+### **Managing Token Transferability**
+
+The manager of an ESDT token can restrict transferability by setting **ESDTTransferRole** to specific addresses. 
+
+When no address has the role, the token can be transferred freely. If at least one address has the role, only addresses with the role can send or receive the token. An address without the role can still transfer tokens to an address with the role.
```
Comment (bogdan-rosianu) [inline docs/tokens/esdt-tokens.md:631]: >If at least one address has the role, only addresses with the role can send or receive the token. An address without the role can still transfer tokens to an address with the role.

a bit contradictory. "only addresses with the role can send or receive" vs "An address without the role can still transfer"

File: docs/tokens/esdt-tokens.md [hunk 2]
```diff
@@ -624,6 +624,30 @@ RolesAssigningTransaction {
...
+:::tip
+Previously, on Mainnet, there was a limitation where the sender and recipient of ESDT tokens had to be in the same shard. 
+Starting from epoch 795, the implementation has been updated to remove this restriction by setting ESDTTransferRole. However, tokens that had transferability set before this epoch will require an additional transaction to follow the latest implementation.
+
+```rust
+SendAllTransferRoleAddressesTransaction {
+    Sender: <address of the ESDT manager>
+    Receiver: erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u
+    Value: 0
+    GasLimit: 100000000
```
Comment (bogdan-rosianu) [inline docs/tokens/esdt-tokens.md:644]: is 100 mil ok?

File: docs/tokens/esdt-tokens.md [hunk 3]
```diff
@@ -624,6 +624,30 @@ RolesAssigningTransaction {
...
+    Data: "sendAllTransferRoleAddresses" +
+          "@" + <token identifier in hexadecimal encoding> +
+}
+```
+:::
```
Comment (bogdan-rosianu) [inline docs/tokens/esdt-tokens.md:649]: append this line under the transaction box:
> _For more details about how arguments have to be encoded, check [here](/developers/sc-calls-format)._

File: docs/tokens/nft-tokens.md [hunk 4]
```diff
@@ -264,14 +264,12 @@ For NFT:
 - ESDTRoleNFTBurn : this role allows one to burn quantity of a specific NFT
 - ESDTRoleNFTUpdateAttributes : this role allows one to change the attributes of a specific NFT
 - ESDTRoleNFTAddURI : this role allows one add URIs for a specific NFT
-- ESDTTransferRole : this role enables transfer only to specified addresses. The owner of the NFT and the address with the ESDTTransferRole should be located on the same shard. The addresses with the transfer role can transfer anywhere.
```
Comment (bogdan-rosianu) [inline docs/tokens/nft-tokens.md:267]: why did you remove this? this is still a valid role. suggestion to update the description and link to the newly added chapter

File: docs/tokens/nft-tokens.md [hunk 5]
```diff
@@ -264,14 +264,12 @@ For NFT:
...
 
 For SFT:
 
 - ESDTRoleNFTCreate : this role allows one to create a new SFT
 - ESDTRoleNFTBurn : this role allows one to burn quantity of a specific SFT
 - ESDTRoleNFTAddQuantity : this role allows one to add quantity of a specific SFT
-- ESDTTransferRole : this role enables transfer only to specified addresses. The owner of the SFT and the address with the ESDTTransferRole should be located on the same shard. The addresses with the transfer role can transfer anywhere.
```
Comment (bogdan-rosianu) [inline docs/tokens/nft-tokens.md:274]: same here

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: docs/tokens/esdt-tokens.md (hunk 3 above)
Comment (dragos-rebegea) [reply docs/tokens/esdt-tokens.md:649]: done

File: docs/tokens/esdt-tokens.md (hunk 2 above)
Comment (dragos-rebegea) [reply docs/tokens/esdt-tokens.md:644]: 50 mil is necessary. Lowered to 60 mil as discussed

File: docs/tokens/esdt-tokens.md (hunk 1 above)
Comment (dragos-rebegea) [reply docs/tokens/esdt-tokens.md:631]: edited

File: docs/tokens/esdt-tokens.md [hunk 6]
```diff
@@ -624,6 +624,30 @@ RolesAssigningTransaction {
 
 _For more details about how arguments have to be encoded, check [here](/developers/sc-calls-format)._
 
+### **Managing Token Transferability**
```
Comment (ovidiuolteanu) [inline docs/tokens/esdt-tokens.md:627]: Can you also add a line after 581 with:
- **ESDTTransferRole**: this role restricts transferability of the token only to the addresses that have the role set, while these addresses can send to any address

**Review by ovidiuolteanu — CHANGES_REQUESTED** [review]

File: docs/tokens/nft-tokens.md (hunk 4 above)
Comment (ovidiuolteanu) [reply docs/tokens/nft-tokens.md:267]: We need to leave this in, but only remove this part:
"The owner of the NFT and the address with the ESDTTransferRole should be located on the same shard. "

File: docs/tokens/nft-tokens.md (hunk 5 above)
Comment (ovidiuolteanu) [reply docs/tokens/nft-tokens.md:274]: We need to leave this in, but only remove this part:
"The owner of the NFT and the address with the ESDTTransferRole should be located on the same shard. "

**Review by ovidiuolteanu — CHANGES_REQUESTED** [review]

File: docs/tokens/esdt-tokens.md [hunk 7]
```diff
@@ -624,6 +624,30 @@ RolesAssigningTransaction {
...
+
+The manager of an ESDT token can restrict transferability by setting **ESDTTransferRole** to specific addresses. 
+
+When no address has the role, the token can be transferred freely. If at least one address has the role, only addresses with the role can send or receive the token. An address without the role can still transfer tokens to an address with the role.
+
+When the role is removed from all addresses, the token will be transferable again with no restrictions.
+
+:::tip
+Previously, on Mainnet, there was a limitation where the sender and recipient of ESDT tokens had to be in the same shard. 
+Starting from epoch 795, the implementation has been updated to remove this restriction by setting ESDTTransferRole. However, tokens that had transferability set before this epoch will require an additional transaction to follow the latest implementation.
```
[the remark this answers is not in the document]
Comment (dragos-rebegea) [reply docs/tokens/esdt-tokens.md:637]: done

File: docs/tokens/esdt-tokens.md (hunk 6 above)
Comment (dragos-rebegea) [reply docs/tokens/esdt-tokens.md:627]: done

File: docs/tokens/nft-tokens.md (hunk 4 above)
Comment (dragos-rebegea) [reply docs/tokens/nft-tokens.md:267]: only removed this part:
"The owner of the NFT and the address with the ESDTTransferRole should be located on the same shard. "

File: docs/tokens/nft-tokens.md (hunk 5 above)
Comment (dragos-rebegea) [reply docs/tokens/nft-tokens.md:274]: only removed this part:
"The owner of the NFT and the address with the ESDTTransferRole should be located on the same shard. "

**Review by bogdan-rosianu — APPROVED** [review]

**Review by ovidiuolteanu — APPROVED** [review]
